clang/lib/Frontend/CompileJobCache.cpp

```diff
@@ -293,11 +300,27 @@ std::optional<int> CompileJobCache::initialize(CompilerInstance &Clang) {
   if (!CAS || !Cache)
     return 1; // Exit with error!
 
+  CompilerInvocation DummyInvocation = Invocation;
+  CompileJobCachingOptions DummyCacheOpts;
+
+  std::string CASInputFileID;
+  std::swap(CASInputFileID, Invocation.getFrontendOpts().CASInputFileID);
   CompileJobCachingOptions CacheOpts;
   ResultCacheKey =
       canonicalizeAndCreateCacheKey(*CAS, Diags, Invocation, CacheOpts);
   if (!ResultCacheKey)
     return 1; // Exit with error!
+  std::swap(CASInputFileID, Invocation.getFrontendOpts().CASInputFileID);
+
+  std::string CASInputFileCacheKey;
+  std::swap(CASInputFileCacheKey,
+            DummyInvocation.getFrontendOpts().CASInputFileCacheKey);
+  ResultCacheKeyWithInputCacheKeysResolved = canonicalizeAndCreateCacheKey(
```

Comment:
We should skip creating a second key in the common case where the cas input file is empty and just copy the other key.